internal/redis-db/redis_test.go

```diff
@@ -71,12 +71,12 @@ func TestNewRedisClient(t *testing.T) {
 		{
 			name:      "single address",
 			addresses: []string{"localhost:6379"},
-			wantErr:   false,
```

Comment:
I think `wantErr` should be `false`, given that the URL is correct, present, and uses a standalone(single URL).

It'd only be `true` when the Redis instance is not running. 

We can extend the test to add a wrong port or host and expect an error(the ping should throw an error)

Reply:
The issue here is if Redis is not set up already, and you run the test case, the test case itself will panic as it will be trying to assert on NoError. This is the reason why I had made wantErr `true`. Could you let me know what is to be done instead in such cases?

Reply:
```suggestion
		{
			name:      "single address with wrong port",
			addresses: []string{"localhost:63749"},
			wantErr:   true,
		},
```

We can add one more test to simulate Redis not being up or ready (using an incorrect port). This should cover your edge case